Fix: Prevent note-card overflow & overlap (#507)

Vaibhav701161 opened the pull request:
fixes #507 

This PR introduces a cleaner, CSS-first approach to layout and enhances test coverage with a focused E2E check to prevent regressions.

1. Layout Overhaul

Replaced custom JS-based absolute positioning with a responsive CSS Grid (auto-fit-notes).

Removed calculateGridLayout, calculateMobileLayout, resize listeners, and related state management.

Cards now scale naturally (min-h-[200px] h-fit) without overlapping.

2. NoteCard Enhancements

Refactored card internals to use Flexbox, ensuring checklist items expand vertically as needed.

Eliminated fixed height calculations—overflow is handled gracefully by design.

E2E Test:
Added tests/e2e/note-card-layout.spec.ts – one concise scenario:
Edit a checklist item with very long content.
Assert card shows the text and scrollHeight === clientHeight (no overflow).

<img width="428" height="116" alt="Screenshot 2025-08-19 112634" src="https://github.com/user-attachments/assets/aed2bd84-39cd-42c6-9354-bb482358c1c7" />

after fix :

https://github.com/user-attachments/assets/cb51ce82-c12d-4249-808a-b47decb68968

Vaibhav701161 commented:
@slavingia Please review . I had ensured zero technical debt i.e. no new states or listeners were added !

Anshgrover23 reviewed (changes requested):
@Vaibhav701161 Tech debt: number of LOC going down, not up.

Vaibhav701161 commented:
Hey @Anshgrover23 — appreciate you pointing that out . Totally agree, tech debt should mean fewer lines and less complexity, not more.

Just to clarify: this PR actually removes a lot more code than it adds. 

I had removed old layout functions (calculateGridLayout, calculateMobileLayout, resize listeners, positioning math, etc.)

Added mostly CSS Grid utility in globals.css and a couple of flexbox tweaks in components

The key here is we deleted the custom bin-packing logic + manual resize handling, and replaced it with plain CSS Grid/Flexbox. That means:

Less code to maintain

No performance-heavy resize listeners

No brittle height/positioning math

Browsers handle responsiveness natively

So the bug fix comes with a leaner, simpler codebase exactly the kind of tech debt cleanup we want .
